apps/ledger-live-mobile/src/screens/Exchange/ProviderList.js

```diff
@@ -0,0 +1,256 @@
+    icon: string,
+    name: string,
+  ) => void,
+};
+
+const ProviderItem = ({ provider, onClick }: ProviderItemProps) => {
+  const [displayedPMs, setDisplayedPMs] = useState<string[]>(
+    provider.paymentProviders.slice(0, 4),
+  );
+  const { colors } = useTheme();
+  const manifest = useRemoteLiveAppManifest(provider.appId);
+
+  const onItemClick = useCallback(() => {
+    onClick(provider, manifest.icon, manifest.name);
+  }, [provider, manifest, onClick]);
+
+  const onMorePMsClick = useCallback(() => {
+    setDisplayedPMs([...provider.paymentProviders]);
+  }, [provider.paymentProviders]);
+
+  if (!manifest) {
+    return null;
+  }
+
+  return (
+    <TouchableOpacity
+      onPress={onItemClick}
+      style={[
+        styles.itemRoot,
+        {
+          backgroundColor: colors.card,
+          borderColor: colors.border,
+        },
+      ]}
+    >
+      <TrackScreen category="Multibuy" name="ProviderList" />
+      <View style={styles.itemLeft}>
+        <View style={styles.itemHeader}>
+          <AppIcon icon={manifest.icon} name={manifest.name} size={32} />
+          <LText secondary style={styles.headerLabel}>
+            {manifest.name}
+          </LText>
+        </View>
+        <View style={styles.pmsRow}>
+          {displayedPMs.map(paymentProvider => (
+            <View
+              key={paymentProvider}
+              style={[styles.pm, { borderColor: colors.border }]}
+            >
+              {assetMap[paymentProvider] ? (
+                assetMap[paymentProvider]({})
+              ) : (
+                <LText style={styles.pmLabel}>{paymentProvider}</LText>
+              )}
+            </View>
+          ))}
+          {provider.paymentProviders.length > 4 && displayedPMs.length <= 4 && (
+            <TouchableOpacity
+              onPress={onMorePMsClick}
+              style={[styles.pm, { borderColor: colors.border }]}
+            >
+              <LText style={styles.pmLabel}>
+                +{provider.paymentProviders.length - 4} more
+              </LText>
+            </TouchableOpacity>
+          )}
+        </View>
+      </View>
+      <ArrowRight size={16} color={colors.grey} />
+    </TouchableOpacity>
+  );
+};
+
+type Props = {
+  navigation: any,
+  route: { params: RouteParams, name: string },
+};
+
+type RouteParams = {
+  accountId: string,
+  accountAddress: string,
+  currency: CryptoCurrency | TokenCurrency,
+  type: "onRamp" | "offRamp",
+};
+
+export default function ProviderList({ route }: Props) {
+  const navigation = useNavigation();
+  const { t } = useTranslation();
+  const { colors } = useTheme();
+  const rampCatalog = useRampCatalog();
+  const { currency, type, accountId, accountAddress } = route.params;
+  const fiatCurrency = useSelector(counterValueCurrencySelector);
+
+  const filteredProviders = filterRampCatalogEntries(rampCatalog.value[type], {
+    cryptoCurrencies: currency.id ? [currency.id] : undefined,
```

Comment:
both tokencurrency and cryptocurrency have an `.id` so this if can be simplified